app/src/test/java/org/simple/clinic/instantsearch/InstantSearchUpdateTest.kt

```diff
@@ -260,4 +263,41 @@ class InstantSearchUpdateTest {
             hasEffects(SaveNewOngoingPatientEntry(ongoingPatientEntry))
         ))
   }
+
+  @Test
+  fun `when blank bp passport result is register new patient, then register new patient`() {
+    val ongoingPatientEntry = OngoingNewPatientEntry.fromFullName("")
+        .withIdentifier(identifier)
+
+    val facility = TestData.facility(
+        uuid = UUID.fromString("2bd05cc3-5c16-464d-87e1-25b6b1a8a99a")
+    )
+    val facilityLoadedModel = defaultModel
+        .facilityLoaded(facility)
+
+    updateSpec
+        .given(facilityLoadedModel)
+        .whenEvent(BlankBpPassportResultReceived(RegisterNewPatient))
+        .then(assertThatNext(
+            hasNoModel(),
+            hasEffects(SaveNewOngoingPatientEntry(ongoingPatientEntry))
+        ))
+  }
+
+  @Test
+  fun `when blank bp passport result is add to existing patient, then do nothing`() {
```

Comment:
Can we rename these test methods to reflect the business use case rather than the implementation details?